xrpl/wallet/main.py

```diff
@@ -16,69 +17,131 @@ class Wallet:
-        """
-        self.seed = seed
-        """
-        The core value that is used to derive all other information about
-        this wallet. MUST be kept secret!
+            public_key: The public key for the account.
+            private_key: The private key used for signing transactions for the account.
+            master_address: Include if a Wallet uses a Regular Key Pair. It must be
+                the master address of the account. The default is `None`.
+            seed: The seed used to derive the account keys. The default is `None`.
         """
-
-        pk, sk = derive_keypair(self.seed, algorithm=algorithm)
-        self.public_key = pk
+        self.public_key = public_key
         """
         The public key that is used to identify this wallet's signatures, as
         a hexadecimal string.
         """
 
-        self.private_key = sk
+        self.private_key = private_key
         """
         The private key that is used to create signatures, as a hexadecimal
         string. MUST be kept secret!
         """
 
-        self.classic_address = derive_classic_address(self.public_key)
-        """The address that publicly identifies this wallet, as a base58 string."""
+        self._classic_address = (
+            ensure_classic_address(master_address)
+            if master_address is not None
+            else derive_classic_address(self.public_key)
+        )
+        """Internal variable for classic_address. Use classic_address instead."""
 
-        self.sequence = sequence
+        self.seed = seed
         """
-        The next available sequence number to use for transactions from this
-        wallet.
-        Must be updated by the user. Increments on the ledger with every successful
-        transaction submission, and stays the same with every failed transaction
-        submission.
+        The core value that is used to derive all other information about
+        this wallet. MUST be kept secret!
         """
 
     @classmethod
-    def create(
-        cls: Type[Wallet], crypto_algorithm: CryptoAlgorithm = CryptoAlgorithm.ED25519
+    def generate(
+        cls: Type[Wallet], algorithm: CryptoAlgorithm = CryptoAlgorithm.ED25519
     ) -> Wallet:
         """
         Generates a new seed and Wallet.
 
         Args:
-            crypto_algorithm: The key-generation algorithm to use when generating the
-                seed. The default is Ed25519.
+            algorithm: The digital signature algorithm to generate an address for.
+                The default is Ed25519.
 
         Returns:
             The wallet that is generated from the given seed.
         """
-        seed = generate_seed(algorithm=crypto_algorithm)
-        return cls(seed, sequence=0, algorithm=crypto_algorithm)
+        seed = generate_seed(algorithm=algorithm)
+        return Wallet.from_seed(seed, algorithm=algorithm)
+
+    @classmethod
+    def from_seed(
+        cls: Type[Wallet],
+        seed: str,
+        *,
```

Comment:
Why does this have a * parameter?

Reply:
(Similar question for the functions below)

Reply:
* in Python is used to tell program that params after * need to be named when called like from_seed(seed, algorithm=algorithm) instead of from_seed(seed, None, algorithm)

I added * because params like master_address and algorithm are like options passed into the method. Instead of passing in opts param, I used *. I think it's clearer than having no * since the order of parameters would matter then.

Reply:
Might be worth adding to the docstring for the functions that use it because it seems like a rare enough feature that we wouldn't expect new coders to know it before encountering our library.

Reply:
I think it might be okay because if they write `Wallet.from_seed(seed, algorithm)`, program will error with something like "TypeError: from_seed() takes 1 positional arguments but 2 were given" which is easily debuggable + googleable.